virtual_rainforest/core/grid.py

```diff
@@ -29,12 +33,22 @@
 """
 
 
+class GridStructure(NamedTuple):
+    """Data structure to be returned from grid creator functions.
+
+    The contents should be two equal length lists giving unique integer ID values
+    and corresponding shapely.geometry.Polygon objects.
+    """
+
+    ids: list[int]
+    polygons: list[Polygon]
```

Comment:
Given that you need to ensure they have the same length, I would use a `dataclass` instead, which lets you do validation afterwards. Moreover, I will make it `frozen` and use `tuples` instead of `lists` if they are not meant to be modified afterwards. Something like:

```suggestion
from dataclasses import dataclass
...
@dataclass(frozen=True)
class GridStructure:
    """Data structure to be returned from grid creator functions.

    The contents should be two equal length lists giving unique integer ID values
    and corresponding shapely.geometry.Polygon objects.
    """

    ids: Tuple[int, ...]
    polygons: Tuple[Polygon, ...]
    
    def __post_init__(self):
        if len(self.ids) != len(self.polygons):
            raise ValueError("Useful description of error...")
```

Reply:
Yup - that makes a lot of sense. These should indeed be frozen and equal length.